Fabric.Identity.API/scripts/CatalystDosIdentity.psm1

```diff
@@ -0,0 +1,620 @@
+        client_secret = $secret
+    }
+
+    $accessTokenResponse = Invoke-RestMethod -Method Post -Uri $url -Body $body
+    return $accessTokenResponse.access_token
+}
+
+<#
+    .Synopsis
+    Attempts to get an access token for the fabric installer client
+
+    .Description
+    Takes in the identity server url and the installer secret and return the installer access token.
+
+    .Parameter identityUrl
+    The identity url to get the access token from
+
+    .Parameter secret
+    The secret of the fabric installer client
+
+    .Example
+    Get-AccessToken -identityUrl "https://server/identity" -secret "SECrEtStrING"
+#>
+function Get-FabricInstallerAccessToken {
+    param(
+        [Parameter(Mandatory = $True)] [Uri] $identityUrl,
+        [Parameter(Mandatory = $True)] [string] $secret
+    )
+
+    $clientId = "fabric-installer"
+    $scope = "fabric/identity.manageresources fabric/authorization.read fabric/authorization.write fabric/authorization.dos.write fabric/authorization.manageclients"
+    return Get-AccessToken $identityUrl $clientId $secret $scope
+}
+
+<#
+    .Synopsis
+    Attempts to retrive information about an existing identity client
+
+    .Description
+    Takes in the identity server url, the clientid and an access token.
+    Returns a client identity object
+
+    .Parameter identityUrl
+    The base identity url
+
+    .Parameter clientId
+    the identifier for the client to retrieve
+
+    .Parameter accessToken
+    an access token previously retrieved from the identity server
+
+    .Example
+    Get-ClientRegistration -identityUrl "https://server/identity" -clientId "sample-client-id" -accessToken "eyJhbGciO"
+#>
+function Get-ClientRegistration {
+    param(
+        [Parameter(Mandatory = $True)] [Uri] $identityUrl,
+        [Parameter(Mandatory = $True)] [string] $clientId,
+        [Parameter(Mandatory = $True)] [string] $accessToken
+    )
+
+    [Uri]$url = "$($identityUrl.OriginalString)/api/v1/client/$clientId"
+
+    $headers = @{"Accept" = "application/json"}
+    $headers.Add("Authorization", "Bearer $accessToken")
+
+    $clientResponse = Invoke-RestMethod -Method Get -Uri $url -Headers $headers
+    return $clientResponse
+}
+
+<#
+    .Synopsis
+    Attempts to register a new identity client
+
+    .Description
+    Takes in the identity server url, the json client body and an access token.
+    Returns a client secret for the new client.
+    If the client already exists, this function updates the client attributes and resets the client secret.
+
+    .Parameter identityUrl
+    The base identity url
+
+    .Parameter body
+    the json payload of attributes and values for the new client
+
+    .Parameter accessToken
+    an access token previously retrieved from the identity server
+
+    .Example
+    New-ClientRegistration -identityUrl "https://server/identity" -body @'{"clientId":"fabric-installer", "clientName":"Fabric Installer" ...... }@' -accessToken "eyJhbGciO"
```

Comment:
The @'s are for multiline strings right? I don't think those need to be here

Reply:
@ are for structures, but in this case we are expecting a single json string, so still not necessary.  Although powershell will auto-convert the hasttable @{} into a json string for us.

Reply:
you are correct, removed